src/rgw/rgw_rest_sts.cc

```diff
@@ -75,6 +75,15 @@ void RGWREST_STS::send_response()
 
 int RGWSTSGetSessionToken::verify_permission()
 {
+  rgw::IAM::Partition partition = rgw::IAM::Partition::aws;
+  rgw::IAM::Service service = rgw::IAM::Service::s3;
+  if (!verify_user_permission(this,
+                              s,
+                              rgw::IAM::ARN(partition, service, "", s->user->user_id.tenant, ""),
+                              rgw::IAM::stsGetSessionToken)) {
+    return -EACCES;
+  }
```

Comment:
And everything comes together.

Reply:
yes this is nice

Reply:
@mattbenjamin : I have updated the PR and have tested the same. It works fine. Before merging I wanted to highlight that currently in order to attach a user policy to control STS authentication, a client will have to use s3curl or something of the sort. In order to use boto, we need to add another namespace for IAM (like STS) which requires some more time and effort. I hope using s3curl will not be a problem.

Reply:
Is the ultimate goal to be able to specify a hostname to bind the IAM/Policy operations to at top level? If so I don't think it's too bad if we merge this now and make that a separate PR.

Reply:
> Is the ultimate goal to be able to specify a hostname to bind the IAM/Policy operations to at top level? If so I don't think it's too bad if we merge this now and make that a separate PR.

Yes, that would be the way to solve it correctly.